daemon/api_test.go

```diff
@@ -3541,14 +3555,14 @@ func (s *apiSuite) TestInstallFails(c *check.C) {
 	}
 
 	d := s.daemonWithFakeSnapManager(c)
-
+	s.vars = map[string]string{"name": "hello-world"}
```

Comment:
Why did you have to change that?

Reply:
It wasn't mocked before, and the code ended up using `""` as snap name inside. Now with the checks in place, this was caught and the test started failing.